src/components/Personalization/createExecuteCachedViewDecisions.js

```diff
@@ -14,10 +14,12 @@ import isNonEmptyArray from "../../utils/isNonEmptyArray";
 
 export default ({ executeViewDecisions, collect }) => {
   return ({ viewName, viewDecisions }) => {
+    // if there are viewDecisions for current view we will execute them and then send the collect call
     if (isNonEmptyArray(viewDecisions)) {
       executeViewDecisions(viewDecisions);
-      return;
+      return; // return here is to avoid the following code to be executed, that one is meant for the condition when viewDecisions is empty
```

Comment:
Hey @ninaceban, I was hoping for more of the _why_ and less of the _what_. I think the _why_ should answer the question, "_Why_ don't we want to send an event when there are view decisions?" For example, is it because we're sending events inside of `executeViewDecisions` in that case so we don't need to send an event here?

Reply:
Ok, I will change the comment